### DISCUSSION TITLE
How to update session after updating profile details

### DISCUSSION THREAD

**Discussion by drashtibpatel** [nextauthjs/next-auth#4256] [Help]
### Question 💬

Just want to know how to update the next-auth session after updating the profile?

### How to reproduce ☕️

once you successfully logged in to the system just go to the My Profile or Account Details page and change your first name and last name. but after updating the profile I'm not able to update/refresh the next-auth session. It shows old First Name, Last Name, Profile Picture...etc

**Comment by zakaryan2004** [comment]
Would like to know the same

**Comment by ataschz-g** [comment]
I'm not sure if exist a native way to do that... :/ Maybe this comment have the solution? https://github.com/nextauthjs/next-auth/issues/596#issuecomment-943453568

**Comment by Timsctt** [comment]
Hi,
It may exist a cleaner way to do that, but I've solved it by doing like this:

```typescript
// [...nextauth.ts]
export const authOptions: NextAuthOptions = {
  ...
  callbacks: {
    ...
    async session(params: { session: Session; user: User | AdapterUser; token: JWT }) {
      params.session.token = params.token.jwt;

      if (params.token.jwt != null && params.token.jwt != undefined) {
        let updatedUser = await fetch(
          'http://localhost:3000/api/account/me?token=' + params.token.jwt,
          {
            method: 'GET',
          }
        ).then((response) => {
          return response.json();
        });
        params.session.user = {
          ...params.session.user,
          name: updatedUser.username,
          email: updatedUser.email,
          image: updatedUser.image,
        };
      }

      return params.session;
    },
  ...
```

Where `http://localhost:3000/api/account/me?token` is an api route to make a request to have updating user infos.
This means that on every session request it make a call to this route.
